server/terraform/modules/myhealth/main.tf

```diff
@@ -97,11 +98,14 @@ resource "google_firebase_project_location" "fireloc" {
 }
 
 
-# App Engine
+# App Engine - required to create the Firestore database.
+#   See: https://firebase.google.com/docs/firestore/solutions/automate-database-create#create_a_database_with_terraform
+#   NOTE: contrary to this documentation, we seem to still need a manual step to actually create the Firestore database.
+#         That manual step is documented in `server/terraform/README.md`.
 resource "google_app_engine_application" "gae" {
   project       = var.project_id
   location_id   = var.region
-  database_type = "CLOUD_DATASTORE_COMPATIBILITY"
+  database_type = var.backend_version == "v2" ? "CLOUD_FIRESTORE" : "CLOUD_DATASTORE_COMPATIBILITY"
```

Comment:
No need for backward compatibility, so you can remove the v1 version once v2 is working.... and don't need to wait for the v1 server to be turned down.

Reply:
As per Matthew's suggestion, I've split the v1 and v2 files into separate modules.